errdefs/errdefs.go

```diff
@@ -0,0 +1,31 @@
+package errdefs
+
+import "errors"
+
+type internalErr struct {
+	error
```

Comment:
Do we need to add a pointer in the error structs in the containerd package?

The other option would be have these as a sibling in some sort of causal wrapper.

Reply:
Did not fully understand the question, but this error type will return correctly in the current containerd `IsInternal` function.

Iiuc there is a issue that this `customMessage` https://github.com/containerd/errdefs/blob/124d0dc9f0cd27578004a3f14e01ef8a3456d147/errors.go#L396 does not have a `Unwrap` method but otherwise the custom `IsInterface` there looks equivalent to `errors.As()` in here.

Reply:
Right now, all of the errors in errdefs don't have an internal pointer to a cause: https://github.com/containerd/errdefs/blob/124d0dc9f0cd27578004a3f14e01ef8a3456d147/errors.go#L40. I think we are good with the matching interface, but classically, errdefs represent the end of the line for the error chain.

Reply:
> but classically, errdefs represent the end of the line for the error chain.

I don't think that's a common case for internal actually. End error is likely a syscall error/parse error etc. (or if containerd loses the syscall error then the containerd API error). If containerd, for example, says `blob not found,` it could be a user error if the blob is wrong in manifest or wrong user input, but internal if it is during history addition, for example. Eg. in this PR, all the errors that come wrong writing history are automatically internal errors (they could be errors from blobstore, bolt, leases etc) because buildkit guarantees that it can save history on any build, no matter if the user asks for something broken. Same for syscall, `lstat(user-copy-path)` is user error, but `lstat(boltdb)` is internal.

Reply:
Yes, agreed. Most internally errors will be unhandled errors that bubble up.